Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/Workspaces/PruneContentStreams.feature

```diff
@@ -118,28 +122,131 @@ Feature: If content streams are not in use anymore by the workspace, they can be
+
+    When I am in workspace "review" and dimension space point {}
+    Then I expect node aggregate identifier "nody-mc-nodeface" to lead to node review-cs-rebased;nody-mc-nodeface;{}
+    And I expect this node to have the following properties:
+      | Key  | Value           |
+      | text | "Review Edited" |
+    Then I expect node aggregate identifier "nodimus-secondus" to lead to node review-cs-rebased;nodimus-secondus;{}
+    And I expect this node to have the following properties:
+      | Key  | Value     |
+      | text | "Live WS" |
+
+    # content stream is still writeable
+    When the command SetNodeProperties is executed with payload:
+      | Key             | Value                           |
+      | workspaceName   | "review"                        |
+      | nodeAggregateId | "nody-mc-nodeface"              |
+      | propertyValues  | {"text": "Review after replay"} |
+    When I am in workspace "review" and dimension space point {}
+    Then I expect node aggregate identifier "nody-mc-nodeface" to lead to node review-cs-rebased;nody-mc-nodeface;{}
+    And I expect this node to have the following properties:
+      | Key  | Value                 |
+      | text | "Review after replay" |
+
+  Scenario: Pruning removed content streams and replaying will lead to workspaces without content stream (and the workspace not fetch able)
+    When the command CreateWorkspace is executed with payload:
+      | Key                | Value                |
+      | workspaceName      | "user-test"          |
+      | baseWorkspaceName  | "live"               |
+      | newContentStreamId | "user-cs-identifier" |
+
+    When the command RebaseWorkspace is executed with payload:
+      | Key                         | Value                        |
+      | workspaceName               | "user-test"                  |
+      | rebasedContentStreamId      | "user-cs-identifier-rebased" |
+      | rebaseErrorHandlingStrategy | "force"                      |
+
+    Then I expect the content stream "user-cs-identifier" to not exist
+    And I prune removed content streams from the event stream
+    Then I expect exactly 0 events to be published on stream "ContentStream:user-cs-identifier"
+
+    Then I expect the highest sequence number to be 8
+    # replay before the rebase, when the workspaces content stream does not exist
+    And I replay the content graph until 7
+
+    Then I expect the workspace "user-test" to not exist
+    Then I expect the following workspaces to exist:
+      | name   | base | status       | content stream  | publishable changes |
+      | "live" | null | "UP_TO_DATE" | "cs-identifier" | false               |
+
+    When I am in workspace "user-test" and dimension space point {}
+    # FIXME maybe getContentGraph should already throw an exception if the content stream does not exist?
+    Then I expect node aggregate identifier "root-node" to lead to no node
```

Comment:
todo its a little dirty that `getContentGraph` still works in the detached state (as we just fetch the `currentContentStreamId` but dont validate that it exists.

its definitely interesting to think about this as catchup hooks will need to deal with this in case youd want to listen to all `CreateWorkspace` events and get some information at that time from the content graph or the workspace finder ...